redbot/cogs/cleanup/cleanup.py

```diff
@@ -300,7 +328,7 @@ async def after(
     async def before(
         self,
         ctx: commands.Context,
-        message_id: RawMessageIds,
+        message_id: Optional[RawMessageIds],
         number: positive_int,
```

Comment:
```suggestion
        message_id: Optional[RawMessageIds],
        number: positive_int,
```
I don't think this works when the number of messages you want to delete can be interpreted as a message ID.

Reply:
Funnily enough, its fine and Optional just handles it.

Reply:
In order for this to be an issue the number would have to be 17 digits long so it's unlikely you'd ever want to delete that many messages.